Watchman.Discord/Areas/Protection/Services/PunishmentsCachingService.cs

```diff
@@ -71,6 +67,33 @@ public async Task AddUserPunishment(ulong userId, Punishment punishment)
             await this._commandBus.ExecuteAsync(command);
         }
 
+        public async Task ClearAndLoadWarnEventsToCache()
+        {
+            _warnsByServer = new FriendlyDictionary<ulong, FriendlyDictionary<ulong, List<WarnEvent>>>();
+            var query = new GetWarnEventsQuery(serverId: 0, receiverId: 0, from: new DateTime(), to: DateTime.Now);
+            var warns = (await this._queryBus.ExecuteAsync(query)).WarnEvents;
+            foreach (var warn in warns)
+            {
+                this.AddWarnToDictionary(warn);
+            }
+        }
+
+        public int GetWarnsCount(ulong serverId, ulong userId, DateTime since)
+        {
+            return _warnsByServer[serverId][userId].Count;
+        }
+
+        public void AddWarnLocal(ulong grantorId, ulong receiverId, string reason, ulong serverId)
+        {
```

Comment:
a co jeśli taki warn istnieje?
w sensie
podobny opis był dodany przykładowo 2 sekundy wcześniej?
(ktoś przypadkiem wpisał to samo, albo dwie osoby chcą dać warna na to samo w tej samej chwili)

Reply:
Można zrobić że po dostaniu warna, dana osoba nie może otrzymać kolejnego przez 30 czy ileśtam sekund i każdy kto spróbuje dostanie wiadomość ile czasu temu był dany warn oraz za co. Jak będzie chciał dać za co innego to sobie poczeka, a jak chciał dać za to samo to wie ze już nie musi?

Reply:
możemy tak zrobić

Reply:
done